<launch>
  <arg name="config" default="$(find-pkg-share autoware_command_mode_decider)/config/default.param.yaml"/>

  <arg name="~/command_mode/transition/available" default="/system/command_mode/transition/available"/>
  <arg name="~/command_mode/availability" default="/system/command_mode/availability"/>
  <arg name="~/command_mode/status" default="/system/command_mode/status"/>
  <arg name="~/command_mode/request" default="/system/command_mode/request"/>
  <arg name="~/operation_mode/state" default="/system/operation_mode/state"/>
  <arg name="~/operation_mode/change_operation_mode" default="/system/operation_mode/change_operation_mode"/>
  <arg name="~/operation_mode/change_autoware_control" default="/system/operation_mode/change_autoware_control"/>
  <arg name="~/mrm/state" default="/system/fail_safe/mrm_state"/>
  <arg name="~/control_mode/report" default="/vehicle/status/control_mode"/>

  <node pkg="autoware_command_mode_decider" exec="command_mode_decider_node">
    <param from="$(var config)"/>
    <remap from="~/command_mode/transition/available" to="$(var ~/command_mode/transition/available)"/>
    <remap from="~/command_mode/availability" to="$(var ~/command_mode/availability)"/>
    <remap from="~/command_mode/status" to="$(var ~/command_mode/status)"/>
    <remap from="~/command_mode/request" to="$(var ~/command_mode/request)"/>
    <remap from="~/operation_mode/state" to="$(var ~/operation_mode/state)"/>
    <remap from="~/operation_mode/change_operation_mode" to="$(var ~/operation_mode/change_operation_mode)"/>
    <remap from="~/operation_mode/change_autoware_control" to="$(var ~/operation_mode/change_autoware_control)"/>
    <remap from="~/mrm/state" to="$(var ~/mrm/state)"/>
    <remap from="~/control_mode/report" to="$(var ~/control_mode/report)"/>
  </node>
</launch>
